### ISSUE TITLE
`when trying to exec the wrapped shell`: No such file or directory (os error 2)

### ISSUE THREAD

**Issue by TheAyes** [nix-community/NixOS-WSL#593]
## Bug description
Unable to start nixos-wsl due to error (see logs section). I've already looked at similar issues and tried the steps provided there. None of which helped. The distro file I only downloaded today so I don't think it's any old.

## To Reproduce
I'm unsure what exactly caused it. But I were hacking around in my flake/configuration and the restarted the terminal and then it stopped working.

## Logs
```
❯ wsl
when trying to exec the wrapped shell

Caused by:
    No such file or directory (os error 2)
```

## WSL version

```
WSL-Version: 2.3.26.0
Kernelversion: 5.15.167.4-1
WSLg-Version: 1.0.65
MSRDC-Version: 1.2.5620
Direct3D-Version: 1.611.1-81528511
DXCore-Version: 10.0.26100.1-240331-1435.ge-release
Windows-Version: 10.0.22631.3737
```

Filehash: `576F2DDA94B0DDE278CD9EBBDF04B53BD9C174FBA5DD4E98BD9AF227DD97FC24`

**Comment by nzbr** [comment]
If this was after a fresh restart of the distro, it might be fixed by #602

**Comment by mmorley0395** [comment]
@nzbr Sorry- what is the fix here? I pulled down a tarball today and ran commands as documented, but got the same error as OP. Is the solution to wait until the next release? Are there downsides to just pulling the minimal ISO and installing it in a WSL environment instead of this tarball, if that is the case?

**Comment by mmorley0395** [comment]
Actually found the solution through your and others comments in other [issues ](https://github.com/nix-community/NixOS-WSL/issues/516) and [discussions](https://github.com/nix-community/NixOS-WSL/discussions/556). I must've had an older WSL version, as `WSL --version` didn't work. I ran `wsl --update` in Powershell, then `wsl --version` worked and I continued with the installation as normal, and installed NixOS-WSL successfully.
